### PR TITLE
Improve SplitPaneState

### PR DESCRIPTION
**Pull request by vdshb**
Improve `SplitPane` programmatic configuration through changes in `SplitPaneState`.
Add programmatic `SplitPane` programmatic configuration changes to demo.

### CODE REVIEW

**Comment by vdshb** [comment]
BTW, I'm not slightly insist on changing the demo-app. I just thought this change might be benificial. It might be easily reverted.

**Comment by vdshb** [comment]
@m-sasha , could you please review this PR?

**Comment by vdshb** [comment]
@m-sasha , @MatkovIvan What can I do to clarify the state of this PR? It's pretty small (tiny if you are not interested in demo-app changes), but it's ignored for quiet a while.

**Comment by m-sasha** [comment]
Making `moveEnabled` settable makes sense, but I'm not sure what the change to positionPercentage is trying to achieve.
Also, I don't like the code duplication in the `positionPercentage` setter and `dispatchRawMovement`.

**Comment by vdshb** [comment]
@m-sasha, any feedback?

**Comment by vdshb** [comment]
@m-sasha, @MatkovIvan What can I do to make this PR closer to merge? It's become pretty simple and small since the last change.

**Comment by m-sasha** [comment]
Sorry, I've just been busy with more urgent issues recently. I'll try to give it another look soon.

File: components/SplitPane/demo/src/jvmMain/kotlin/org/jetbrains/compose/splitpane/demo/Main.kt [hunk 1]
```diff
@@ -24,40 +41,69 @@ fun main() = singleWindowApplication(
...
+        var percentage by remember { mutableStateOf("0.20") }
+        Row {
+            Column(verticalArrangement = Arrangement.spacedBy(10.dp), modifier = Modifier.padding(10.dp).width(180.dp)) {
+                Text("Action panel", fontWeight = FontWeight.Bold)
+                Button(onClick = { splitterState.moveEnabled = !splitterState.moveEnabled }) {
+                    Text(if (splitterState.moveEnabled) "Freeze V" else "Unfreeze V")
+                }
+                Button(onClick = { hSplitterState.moveEnabled = !hSplitterState.moveEnabled }) {
+                    Text(if (hSplitterState.moveEnabled) "Freeze H" else "Unfreeze H")
+                }
```
Comment (m-sasha) [inline components/SplitPane/demo/src/jvmMain/kotlin/org/jetbrains/compose/splitpane/demo/Main.kt:54]: These could be checkboxes instead.

**Review by m-sasha — APPROVED** [review]

**Review by igordmn — APPROVED** [review]

File: components/SplitPane/demo/src/jvmMain/kotlin/org/jetbrains/compose/splitpane/demo/Main.kt (hunk 1 above)
Comment (igordmn) [reply components/SplitPane/demo/src/jvmMain/kotlin/org/jetbrains/compose/splitpane/demo/Main.kt:54]: Looks fine either way

Comment (vdshb) [reply components/SplitPane/demo/src/jvmMain/kotlin/org/jetbrains/compose/splitpane/demo/Main.kt:54]: I've just find time to finally improve it. But yeah, looks fine either way for me as well.

**Comment by vdshb** [comment]
Thanks for eventually merged improvement!

**Pull request merged**